project-base/storefront/components/Pages/App/AppPageContent.tsx

```diff
@@ -33,6 +34,10 @@ export const AppPageContent: FC<AppPageContentProps> = ({ Component, pageProps }
     const { url } = pageProps.domainConfig;
     const userConsent = usePersistStore((store) => store.userConsent);
 
+    useEffect(() => {
+        usePersistStore.persist.rehydrate();
+    }, []);
```

Comment:
I am not sure if I like this, since it's one more function call for 3 lines of code. But ok, if you prefer it this way, I don't have problem with this approach. Refactored.